updater/bootstrap.ts

```diff
@@ -182,53 +167,39 @@ async function fetchUpdater(info: IUpdateInfo): Promise<string | null> {
     updaterName = 'preview-updater.exe';
   }
 
-  const reqInfo = {
-    baseUrl: info.baseUrl,
-    uri: `/${updaterName}`,
-  };
   const updaterPath = path.resolve(info.tempDir, updaterName);
   if (await isUpdaterRunning(updaterPath, updaterName)) {
     console.log('Updater is already running, aborting fetch.');
     return null;
   }
 
   const outStream = fs.createWriteStream(updaterPath);
-
   return new Promise((resolve, reject) => {
-    const outPipe = request(reqInfo)
-      .on('response', response => {
-        if (response.statusCode !== 200) {
-          reject(`Failed to fetch updater: status ${response.statusCode}`);
-        }
-      })
-      .pipe(outStream);
-
-    outPipe.on('close', () => {
-      resolve(updaterPath);
-    });
-
-    outPipe.on('error', error => {
-      reject(error);
+    fetch(`${info.baseUrl}/${updaterName}`).then(response => {
```

Comment:
If the promise returned by `fetch` rejects, then the outer promise never resolves or rejects.  It's the same bug that you fixed with the `downloadFile` infinite loader.

Reply:
Good catch